### PR TITLE
safety-cli: init at 0.5.1

### PR DESCRIPTION
**Pull request by thomasdesr**
###### Motivation for this change

Makes it easier to check for known vulnerabilities in a projects' python dependencies

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [n/a] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dparse/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  propagatedBuildInputs = [ toml pyyaml ];
+
+  checkInputs = [ pytest ];
+
+  meta = with lib; {
+    description = "A parser for Python dependency files";
+    homepage = "https://github.com/pyupio/safety";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (jonringer) [inline pkgs/development/python-modules/dparse/default.nix:20]: Please make a separate commit adding yourself as a maintainer:

````
maintainers: add thomasdesr
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -26487,4 +26487,6 @@ in
   vpsfree-client = callPackage ../tools/virtualization/vpsfree-client {};
 
   gpio-utils = callPackage ../os-specific/linux/kernel/gpio-utils.nix { };
+
+  safety-cli = with python3.pkgs; toPythonApplication safety;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:26491]: `all-packages.nix` is in disarry, however, please attempt to place this in a more logical location

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -5661,6 +5661,10 @@ in {
 
   safe = callPackage ../development/python-modules/safe { };
 
+  dparse = callPackage ../development/python-modules/dparse { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:5664]: please place dparse in a somewhat more sorted location ;)

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by thomasdesr** [comment]
Thanks for the review, ptal :D

File: pkgs/development/python-modules/dparse/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi, pytest, toml, pyyaml }:
```
Comment (jonringer) [inline pkgs/development/python-modules/dparse/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, isPy27, pytest, toml, pyyaml }:
```

File: pkgs/development/python-modules/dparse/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+buildPythonPackage rec {
+  pname = "dparse";
+  version = "0.5.1";
```
Comment (jonringer) [inline pkgs/development/python-modules/dparse/default.nix:5]:
```suggestion
  version = "0.5.1";
  disabled = isPy27;
```
```
builder for '/nix/store/ai1641qc34fv4zh0bpr1bn22piq3lb52-python2.7-dparse-0.5.1.drv' failed with exit code 1; last 10 log lines:
  adding 'dparse-0.5.1.dist-info/top_level.txt'
  adding 'dparse-0.5.1.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/dparse-0.5.1/dist /build/dparse-0.5.1
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  Processing ./dparse-0.5.1-py2-none-any.whl
  ERROR: Package 'dparse' requires a different Python: 2.7.18 not in '>=3.5'
cannot build derivation '/nix/store/v370y4sdsywpivj56zdf7isj74ypcxgm-python2.7-safety-1.9.0.drv': 1 dependencies couldn't be built
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM, I know I said that already :)

**Comment by thomasdesr** [comment]
Fixed!

ptal! :D

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 87163 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  - python37Packages.dparse  - safety-cli (python37Packages.safety)  - python38Packages.dparse  - python38Packages.safety</details>
